azure/petctl.py

```diff
@@ -0,0 +1,214 @@
+import argparse
+import os
+import subprocess
+import yaml
+import json
+
+# +
+def configure_yaml(yaml_file, petctl_dir,args):
+    
+    print('Configuring azure-pytorch-elastic.yaml....')
+    
+    with open(yaml_file) as f:
+        data = yaml.load(f, Loader=yaml.FullLoader)
+    
+    data["spec"]["parallelism"] = args.max_size
+    data["spec"]["template"]["spec"]["containers"][0]["env"].extend([dict([('name', 'JOB_ID'), ('value', args.name)]),
+                                                                     dict([('name', 'MIN_SIZE'), ('value', str(args.min_size))]),
+                                                                     dict([('name', 'MAX_SIZE'), ('value', str(args.max_size))])])
+    result_yaml = open(os.path.join(petctl_dir, "azure-pytorch-elastic.yaml"),"w")
+    yaml.dump(data,result_yaml)
+
+    
+def configure_json(json_file, petctl_dir,args):
+    
+    print('Configuring kubernetes.json file...')
+    
+    with open(json_file) as f:
+        data =  json.load(f)
+    data["properties"]["masterProfile"]["count"] = 1
+    data["properties"]["agentPoolProfiles"][0]["count"] = args.max_size
+    
+    result_json = open(os.path.join(petctl_dir,"kubernetes.json"),"w")
+    json.dump(data, result_json, indent=4)
+    
+    
+# Create a YAML job file based on user inputs
+def configure(args):
+    PETCTL_DIR             = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__) + 'config/'))
+    SAMPLE_YAML_FILE       = os.path.join(PETCTL_DIR, "sample_specs.yaml")
+    KUBERNETES_JSON_FILE = os.path.join(PETCTL_DIR, "kubernetes.json")
+    
+    configure_yaml(SAMPLE_YAML_FILE, PETCTL_DIR, args)
+    configure_json(KUBERNETES_JSON_FILE, PETCTL_DIR, args)
+    
+    
+
+# +
+def run_job():
+    subprocess.call("./run_job.sh", shell=True)
+
+def setup(args):
+    subprocess.call("./setup.sh " + 
```

Comment:
Probably better not to create this huge string with parameters to setup.sh (which is also a potential security risk because of possible special characters), but have it all in Python and only shell out specific individual commands.